### DISCUSSION TITLE
OVIS-4 vs master

### DISCUSSION THREAD

**Discussion by oceandlr** [ovis-hpc/ldms#653] [General]
@tom95858 @valleydlr 

Can you put here the expectations of OVIS-4 vs master? 

For example: 
* Is master the next major version under development? 
* Are there expected compatibility and API changes for stores and samplers under V4 vs master? If so, how should we handle our contributions (e.g., push a v4 version and then expect to do a master version later once that infrastructure is announced as getting ready for release)?
* What is the timeline? If I wanted to get my store or sampler into something that is going to be an imminent release, should I be adding it to V4 vs master?

**Comment by tom95858** [comment]
On Wed, Mar 3, 2021 at 10:04 PM oceandlr <notifications@github.com> wrote:

> @tom95858 <https://github.com/tom95858> @valleydlr
> <https://github.com/valleydlr>
>
> Can you put here the expectations of OVIS-4 vs master?
>
> For example:
>
>    - Is master the next major version under development?
>
>
Master is going away. All features are being backported to OVIS-4.


>    - Are there expected compatibility and API changes for stores and
>    samplers under V4 vs master? If so, how should we handle our contributions
>    (e.g., push a v4 version and then expect to do a master version later once
>    that infrastructure is announced as getting ready for release)?
>
>
My goal at present is to retain backward compatibility support for the
existing plugin interface and support both types going forward (i.e.
single-instance vs. multi-instance) New samplers can be written to either
interface, but new the multi-instance interface is preferred.


>    - What is the timeline? If I wanted to get my store or sampler into
>    something that is going to be an imminent release, should I be adding it to
>    V4 vs master?
>
>
Nothing should be going to master. OVIS-4 is the default branch. We could
perhaps rename master to something else to avoid confusing those who assume
that master is always the TOT.

Tom



>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ovis-hpc/ovis/discussions/653>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVTPXHWKU62NCMMBMQJPV3TB4IEHANCNFSM4YSRCCJA>
> .
>


-- 
Thomas Tucker, President, Open Grid Computing, Inc.

**Comment by morrone** [comment]
We are now using "main" for the main development branch and "b4.4" style branches for modifications to previous releases. This dicussion is now moot.
